self_hosted/typecheck.jou

```diff
@@ -60,13 +60,13 @@ def fail_with_implicit_cast_error(location: Location, template: byte*, from: Typ
     while *template != '\0':
         if starts_with(template, "<from>"):
             template = &template[6]
-            strcat(message, &from->name[0])
+            strcat(message, from->name)
         elif starts_with(template, "<to>"):
             template = &template[4]
-            strcat(message, &to->name[0])
+            strcat(message, to->name)
         else:
             s = [*template++, '\0']
-            strcat(message, &s[0])
+            strcat(message, s)
```

Comment:
Without the explicit `&` operator, it will probably feel weird to many people why you need to place the array to a temporary variable. For now I should ensure that the error message is clear .